### ISSUE TITLE
chezmoi decryption of all files in separate location to source path

### ISSUE THREAD

**Issue by slurikadre** [twpayne/chezmoi#1449]
## What exactly are you trying to do?

Decrypt all files, without having to specify them individually, in a separate source path.

## What have you tried so far?

```
chezmoi decrypt -S ~/.local/share/chezmoi_encrypted
```

Describe what you have tried so far.

## Where else have you checked for solutions?

* [x] I have read [chezmoi's how-to guide](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md), and not found the answer.
* [x] I have read [chezmoi's FAQ](https://github.com/twpayne/chezmoi/blob/master/docs/FAQ.md), and not found the answer.
* [x] I have searched [chezmoi's reference guide](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md), and not found the answer.
* [x] Other, please give details.

## Additional context

So, I like to keep my encrypted files separate from my source path that I commit to git.

So to encrypt files I normally use this command:

```
chezmoi add --encrypt -S $XDG_DATA_HOME/chezmoi_encrypted $1;
```

What command would I want to run if I want to decrypt all the files?

Will chezmoi know where to decrypt the files?

**Comment by twpayne** [comment]
Running

```console
$ chezmoi apply -S $XDG_DATA_HOME/chezmoi_encrypted
```

will cause chezmoi to decrypt the files you added with `chezmoi add --encrypt -S $XDG_DATA_HOME/chezmoi_encrypted` into their original locations.

To decrypt an individual file you can run:

```console
$ chezmoi cat -S $XDG_DATA_HOME/chezmoi_encrypted ~/path/to/file/in/home/dir
```

You can also get an archive containing all decrypted files, for example:

```console
$ chezmoi archive --output=dotfiles.tar.gz -S $XDG_DATA_HOME/chezmoi_encrypted
```

Hope this helps, please re-open if needed.

**Comment by slurikadre** [comment]
Excellent. I knew it was possible, but hadn't figured it out.

> ```
> --output=dotfiles.tar.gz
> ```

One minor thing there, it appears chezmoi doesn't gzip the archive so .tar is more appropriate.

**Comment by twpayne** [comment]
> One minor thing there, it appears chezmoi doesn't gzip the archive so .tar is more appropriate.

Ah, sorry, the automatic gzip'ing when the output is a `.tar.gz` file was only added in the just-released 2.5.0 version. In earlier versions you can pass the `--gzip` flag.
